Change geometry type in the items table

yellowcap opened the pull request:
**Related Issue(s):** 
https://github.com/stac-utils/stac-fastapi/issues/284

**Description:**
This generalizes the geometry type of the geometry column in the Items table from `Polygon` to `Geometry`.

- As far as I understood, `Polygon` is a subset of `Geometry`, so this migration does not require a data migration step.
- I am not sure about the downgrade migration. It will only work if no other geometries are added to the DB after the change. So maybe there the alternative is simply be a `pass`, as the data in the table might not be backwards compatible. On the other hand, the way its proposed its the proper downgrade from a type perspective.

**PR Checklist:**

- [x] Code is formatted and linted (run `pre-commit run --all-files`)
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/master/CHANGES.md).

lossyrob commented on stac_fastapi/sqlalchemy/alembic/versions/5909bd10f2e6_change_item_geometry_column_type.py:
```diff
@@ -0,0 +1,35 @@
+Create Date: 2021-11-23 10:14:17.974565
+
+"""
+import sqlalchemy as sa
```
Checked linting, just need to remove this line for flake8 to pass

yellowcap replied:
Thanks @lossyrob - just did that and also rebased from current master to avoid conflicts.

yellowcap commented:
I added a copy of a stac item from the L28 L2 collection that has a multipolygon feature (the one that I referenced in the issue https://github.com/stac-utils/stac-fastapi/issues/284). 

With that I added a test for ingesting the item and running a search against it. I am not sure if that is sufficient to properly test the change. So @lossyrob , when you have a moment please have a look and let me know if anything is missing.

yellowcap commented:
:+1: thanks for your help on this @lossyrob
